### DISCUSSION TITLE
[Debugging] Show which datasets are outdated

### DISCUSSION THREAD

**Discussion by francisduval** [kedro-org/kedro-viz#2151] [Ideas]
## Description
When running `kedro viz run`, there is no way to know which datasets are up to date and which ones are outdated. A dataset is said to be outdated if the code upstream has changed since the dataset was run for the last time. This feature exists with the Targets package in R. Also, when you run the targets pipeline, only nodes that are outdated are run, which saves computing time.  

## Context
This could be a nice feature since without it, there is no effective way to tell which parts of the pipeline you should rerun when changes have been made to the code. Sometimes, you are unsure if a dataset is up to date or not, and then you have to rerun it to be sure, which can take a long time. 

## Possible Implementation
Color datasets that are outdated with another color. Also, it would be nice to have a `kedro ` command that would only run outdated datasets, such as `kedro run --only_outdated` or `kedro run --pipeline pipeline_name --only_outdated`.

## Checklist

- [ ] In kedro viz, color outdated datasets with another color.
- [ ] In kedro, include an option which ensures that only outdated datasets are run.

**Comment by astrojuanlu** [comment]
Somewhat related: https://github.com/kedro-org/kedro/issues/221, https://github.com/kedro-org/kedro/issues/2307

**Comment by NeroOkwa** [comment]
Backlog grooming notes:

This was also highlighted in #1750, and would build on the dataset preview and debugging work stream. We should consider implementing this.
Next step - investigation of technical feasibility.

**Comment by astrojuanlu** [comment]
I think showing whether a dataset is outdated or not is tricky, because for remote datasets, you don't have a zero-cost way of computing, say, a hash. So your locally computed hash will become outdated without you noticing.

For the record, this is what Dagster does:

| 1 | 2 | 3 | 4 |
|---|---|---|---|
|  ![image](https://github.com/user-attachments/assets/1a34b271-afde-497f-a27d-8ee70b094d35) | ![image](https://github.com/user-attachments/assets/26febcaf-37c9-46fc-baf0-fe52eb3dc76a)  | ![image](https://github.com/user-attachments/assets/faa4c2f1-c5de-4d7d-81dc-a1c4160f9c37)  | ![image](https://github.com/user-attachments/assets/693fec6e-b82e-4edc-a2b0-3dad6d61ed9d)  |

What do you think @francisduval ?

**Comment by astrojuanlu** [comment]
xref https://github.com/kedro-org/kedro-viz/issues/2310
